src/System.IO.Pipelines/tests/FlushAsyncCompletionTests.cs

```diff
@@ -33,5 +33,33 @@ async Task Await(ValueTask<FlushResult> a)
             Assert.Equal(true, task2.IsFaulted);
             Assert.Equal("Concurrent reads or writes are not supported.", task2.Exception.InnerExceptions[0].Message);
         }
+
+        [Fact]
+        public async Task CompletingWithExceptionDoesNotAffectState()
+        {
+            Pipe.Writer.Complete();
+            Pipe.Writer.Complete(new Exception());
+
+            var result = await Pipe.Reader.ReadAsync();
```

Comment:
Nit, but the tests for ReadAsync are in the FlushAsync file, and the FlushAsync tests in the ReadAsync file.

Reply:
Good catch!

Reply:
I treated FlushAsyncCompletionTests.cs  as writer completion tests and these tests are testing reader and writer completion behaviour.